File: frontend/src/types/approval.ts
// ─── Approval status ─────────────────────────────────────────────────────────

export type ApprovalStatus = "pending" | "approved" | "rejected";

// ─── Contact info on approval card ──────────────────────────────────────────

export interface ApprovalContactInfo {
  id: string;
  full_name: string;
  role: string;
  email?: string | null;
  linkedin_url?: string | null;
  linkedin_confirmed: boolean;
  confidence_score: number;
  email_confidence?: number | null;
  email_verified: boolean;
  source?: string | null;
  is_primary_persona: boolean;
}

// ─── Buying committee ───────────────────────────────────────────────────────

export interface ApprovalCommitteePersona {
  id: string;
  full_name: string;
  role: string;
  linkedin_url: string;
  confidence_score: number;
  source: string;
}

export interface ApprovalBuyingCommittee {
  primary_persona?: ApprovalCommitteePersona | null;
  secondary_personas: ApprovalCommitteePersona[];
  influencers: ApprovalCommitteePersona[];
}

// ─── Full approval item (rich, from API) ─────────────────────────────────────

export interface ApprovalItem {
  id: string;
  recommendation_id: string;
  status: ApprovalStatus;
  comment?: string | null;
  rejection_reason?: string | null;
  decided_at?: string | null;

  // Company context
  company_id: string;
  company_name: string;
  company_domain: string;
  company_industry?: string | null;
  company_logo_url?: string | null;

  // Score & tier
  qualification_score?: number | null;
  tier?: string | null;
  score_breakdown?: Record<string, unknown> | null;

  // Recommendation detail
  priority: string;
  reason: string;
  confidence: number;
  outreach_subject?: string | null;
  outreach_template: string;
  linkedin_message?: string | null;
  market_trigger?: string | null;
  talking_points?: string[] | null;
  buying_committee?: ApprovalBuyingCommittee | null;
  outreach_channel?: string | null;
  follow_up_sequence?: Record<string, unknown>[] | null;

  // Contacts
  contacts: ApprovalContactInfo[];

  // Market signals
  market_signals?: Record<string, unknown> | null;

  // Reviewer
  reviewer_name?: string | null;
}

// ─── Approval stats ─────────────────────────────────────────────────────────

export interface ApprovalStats {
  pending: number;
  approved: number;
  rejected: number;
  avgScore: number;
  highPriority: number;
}

// ─── Bulk action ────────────────────────────────────────────────────────────

export type BulkAction = "approve_all_a" | "reject_all_c" | "approve_selected" | "reject_selected";

// ─── Rejection reasons ──────────────────────────────────────────────────────

export const REJECTION_REASONS = [
  { value: "not_in_target_market", label: "Not in target market" },
  { value: "already_customer", label: "Already a customer" },
  { value: "competitor", label: "Competitor" },
  { value: "too_small", label: "Too small" },
  { value: "too_large", label: "Too large" },
  { value: "wrong_geography", label: "Wrong geography" },
  { value: "bad_email_quality", label: "Bad email quality" },
  { value: "other", label: "Other" },
] as const;

File: backend/schemas/approval.py
"""
Pydantic schemas for approval endpoints.
Extended to return full context per recommendation for the Kanban view.
"""

from __future__ import annotations

from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel


# ── Individual contact on approval card ──────────────────────────────────────


class ApprovalContactInfo(BaseModel):
    id: str
    full_name: str
    role: str
    email: Optional[str] = None
    linkedin_url: Optional[str] = None
    linkedin_confirmed: bool = False
    confidence_score: float = 0.0
    email_confidence: Optional[float] = None
    email_verified: bool = False
    source: Optional[str] = None
    is_primary_persona: bool = False


# ── Buying committee ─────────────────────────────────────────────────────────

class ApprovalCommitteePersona(BaseModel):
    id: str
    full_name: str
    role: str
    linkedin_url: str = ""
    confidence_score: float = 0.0
    source: str = ""


class ApprovalBuyingCommittee(BaseModel):
    primary_persona: Optional[ApprovalCommitteePersona] = None
    secondary_personas: list[ApprovalCommitteePersona] = []
    influencers: list[ApprovalCommitteePersona] = []


# ── Full recommendation detail ────────────────────────────────────────────────


class ApprovalRecommendationDetail(BaseModel):
    id: str
    priority: str
    reason: str
    suggested_action: str
    outreach_template: str = ""
    outreach_subject: Optional[str] = None
    linkedin_message: Optional[str] = None
    confidence: float = 0.0
    buying_committee: Optional[ApprovalBuyingCommittee] = None
    talking_points: Optional[list[str]] = None
    market_trigger: Optional[str] = None
    outreach_channel: Optional[str] = None
    follow_up_sequence: Optional[list[dict]] = None


# ── Score breakdown dimension (for the approval card mini breakdown) ────────


class ApprovalDimensionScore(BaseModel):
    score: float = 0
    weight: float = 0
    detail: str = ""


# ── Full approval response ──────────────────────────────────────────────────


class ApprovalResponse(BaseModel):
    id: str
    recommendation_id: str
    status: str
    comment: Optional[str] = None
    rejection_reason: Optional[str] = None
    decided_at: Optional[datetime] = None

    # Company context
    company_id: str = ""
    company_name: str = ""
    company_domain: str = ""
    company_industry: Optional[str] = None
    company_logo_url: Optional[str] = None

    # Score & tier
    qualification_score: Optional[float] = None
    tier: Optional[str] = None
    score_breakdown: Optional[dict[str, Any]] = None

    # Recommendation detail
    priority: str = ""
    reason: str = ""
    confidence: float = 0.0
    outreach_subject: Optional[str] = None
    outreach_template: str = ""
    linkedin_message: Optional[str] = None
    market_trigger: Optional[str] = None
    talking_points: Optional[list[str]] = None
    buying_committee: Optional[ApprovalBuyingCommittee] = None
    outreach_channel: Optional[str] = None
    follow_up_sequence: Optional[list[dict]] = None

    # Contacts
    contacts: list[ApprovalContactInfo] = []

    # Market signals
    market_signals: Optional[dict[str, Any]] = None

    # Reviewer info
    reviewer_name: Optional[str] = None

    class Config:
        from_attributes = True


# ── Decision submission request ──────────────────────────────────────────────


class ApprovalRequest(BaseModel):
    status: str  # "approved" | "rejected"
    comment: Optional[str] = None
    rejection_reason: Optional[str] = None
    channel: Optional[str] = None       # "email" | "linkedin" | "both"
    scheduled_at: Optional[str] = None  # ISO datetime or "now"
    edited_template: Optional[str] = None
    edited_subject: Optional[str] = None
